crawl-ref/source/exclude.cc

```diff
@@ -128,9 +128,12 @@ void remove_auto_exclude(const monster* mon, bool sleepy)
 travel_exclude::travel_exclude(const coord_def &p, int r,
                                bool autoexcl, string dsc, bool vaultexcl)
     : pos(p), radius(r),
-      los(los_def(p, opc_excl, circle_def(r, C_SQUARE))),
       uptodate(false), autoex(autoexcl), desc(dsc), vault(vaultexcl)
 {
+    if (dsc == "oklob plant")
+        los = los_def(p, opc_solid_see, circle_def(r, C_SQUARE));
+    else 
+        los = los_def(p, opc_excl, circle_def(r, C_SQUARE));
     set_los();
```

Comment:
You could shrink this to 
`los = los_def(p,
              dsc == "oklob_plant" ? opc_solid_see : opc_excl,
              circle_def(r, C_SQUARE));`
if you wanted to. I don't know if that's better or worse, though.